akka-actor-tests/src/test/scala/akka/serialization/SerializationSetupSpec.scala

```diff
@@ -15,13 +18,39 @@ class ProgrammaticDummy
 case class ProgrammaticJavaDummy()
 case class SerializableDummy() // since case classes are serializable
 
+/**
+ * Keeps a registry of each object "serialized", returns an identifier, for every "deserialization" the original object
+ * is returned. Useful for tests needing serialization back and forth but that must avoid java serialization for some
+ * reason.
+ */
+final class FakeSerializer extends Serializer {
+  val includeManifest = false
+  val identifier = 666
+  private val registry = new ConcurrentHashMap[Integer, AnyRef]()
+  private val counter = new AtomicInteger(0)
+
+  def toBinary(o: AnyRef) = {
+    val id = counter.addAndGet(1)
+    require(id < Byte.MaxValue)
+    registry.put(id, o)
+    Array(id.toByte)
+  }
+
+  def fromBinary(bytes: Array[Byte], clazz: Option[Class[_]]) = {
+    require(bytes.length == 1)
+    val id = bytes(0).toInt
+    registry.get(id)
+  }
+}
```

Comment:
A bunch of serialization tests used the TestSerializer, which serializes to empty byte array and then deserializes to null and expected messages to come through.  Renamed to NoopSerializer to make it clear that it throws things away.